File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.py
def solution(arr, flag):
    x = []
    for i in range(len(flag)):
        if flag[i]:
            for _ in range(arr[i] * 2):
                x.append(arr[i])
        else:
            for _ in range(arr[i]):
                x.pop()
    return x
                

File: 프로그래머스/0/181860. 빈 배열에 추가， 삭제하기/빈 배열에 추가， 삭제하기.cpp
#include <vector>

using namespace std;

vector<int> solution(vector<int> arr, vector<bool> flag) {
    vector<int> X;
    
    for (int i = 0; i < arr.size(); ++i) {
        if (flag[i]) {
            int count = arr[i] * 2;
            for (int j = 0; j < count; ++j) {
                X.push_back(arr[i]);
            }
        } else {
            int count = arr[i];
            for (int j = 0; j < count; ++j) {
                X.pop_back();
            }
        }
    }
    
    return X;
}